src/market.rs

```diff
@@ -455,7 +509,43 @@ impl JobState {
                 println!("job {job}: failed to retrieve instance state, {err}");
             }
             Ok(is_running) => {
-                if !is_running && self.rate >= self.min_rate {
+                if is_running {
+                    let is_enclave_running = infra_provider
+                        .check_enclave_running(&self.instance_id, self.region.clone())
+                        .await;
+
+                    match is_enclave_running {
+                        Ok(is_enclave_running) => {
+                            if !is_enclave_running {
+                                println!("job {job}: enclave not running on the instance, running the enclave");
+                                let res = infra_provider
+                                    .run_enclave_on_instance(
+                                        job.encode_hex(),
+                                        &self.instance_id,
+                                        self.region.clone(),
+                                        &self.eif_url,
+                                        self.req_vcpus,
+                                        self.req_mem,
+                                        self.bandwidth,
+                                    )
+                                    .await;
+                                match res {
+                                    Ok(_) => {
+                                        println!(
+                                            "job {job}: enclave successfully ran on the instance"
+                                        );
+                                    }
+                                    Err(err) => {
+                                        println!("job {job}: failed to run enclave, {err}. Spinning instance down.");
```

Comment:
`schedule_termination` here? In case of repeat failure  to run enclave

Reply:
The fn `run_enclave_on_instance` has `spin_down` in it already for every error